src/Npgsql/NpgsqlParameter.cs

```diff
@@ -449,7 +450,11 @@ public sealed override int Size
         /// <inheritdoc />
         [DefaultValue("")]
         [Category("Data")]
-        public sealed override string? SourceColumn { get; set; }
+        public sealed override string? SourceColumn
+        {
+            get => _sourceColumn;
+            set => _sourceColumn = value ?? string.Empty;
```

Comment:
The same here. You lies that the default value is an empty string, because you have never assigned it before calling the getter.

Reply:
We do have this initialization: https://github.com/npgsql/npgsql/blob/a13c9c5213044daf6ee3cbd7e53d8e7ca6ca899d/src/Npgsql/NpgsqlParameter.cs#L59, is there a path where SourceColumn doesn't get called?

Reply:
Haven't seen this. In that case it would be better to coalesce to an empty string in the getter, or set a value in the field definition.

By the way `SourceColumn` sometimes set twice during an object construction.

Reply:
Any particular reason to coalesce in the getter and not in the setter? Are you thinking of some perf gain in not initializing to empty string in the constructor (but it implicitly gets initialized to null instead)? Otherwise it seems cleaner to make sure that the private variable (e.g. _commandText) is non-nullable and coalesce in the setter, no?

Reply:
> By the way SourceColumn sometimes set twice during an object construction.

You're right, I'll fix this.

Reply:
The double-initialization also happens on ParameterName.

Reply:
Not for perf, but for making code less error prone. Setting the default in the definition solves the problem too and probably is more preferable way.

Reply:
Funny, I see it the other way around - better to not get a bad value (null) into your system, but to stop it as early as possible (i.e. when it's set, not when it's gotten). When we complete the nullability work it would be safer too, because it would allow _commandText to actually be non-nullable, which limits potential bugs etc.

Reply:
Feel free to go any way after cleanup since we have the same goal - less bugs (: